### ISSUE TITLE
Integration Test: Conference Workshop Persona (Dr. Patel)

### ISSUE THREAD

**Issue by scttfrdmn** [scttfrdmn/prism#403]
## Objective
Validate rapid deployment of 50 workshop workspaces for a 3-hour conference tutorial with temporary access and auto-termination.

**Persona**: Dr. Patel hosting ACM conference workshop on ML with 50 participants, 3-hour duration, rapid deployment requirement

## Scenario Overview

**ACM Conference Workshop**:
- Participants: 50 attendees
- Duration: 3 hours
- Template: Python Machine Learning (Jupyter notebooks)
- Requirement: Provision all workspaces in < 10 minutes before workshop starts
- Auto-terminate 4 hours after workshop begins
- Generate temporary access credentials

### 1. Pre-Workshop: Rapid Deployment

```go
func TestPersona_ConferenceWorkshop(t *testing.T) {
    if testing.Short() {
        t.Skip("Skipping bulk workshop test in short mode")
    }

    ctx := NewTestContext(t)
    registry := fixtures.NewFixtureRegistry(t, ctx.Client)

    // Create workshop project
    project := fixtures.CreateTestProject(t, registry, fixtures.CreateTestProjectOptions{
        Name:         "acm-ml-workshop-2024",
        Description:  "ACM Conference: Introduction to Machine Learning",
        Budget:       200.00,  // $200 for 3-hour workshop
        BudgetPeriod: "event",
        Owner:        "dr.patel@university.edu",
    })

    // Configure auto-termination (4 hours from now)
    workshopEnd := time.Now().Add(4 * time.Hour)
    setProjectAutoTermination(t, ctx, project.ID, workshopEnd)

    const numParticipants = 50

    // Bulk provision workspaces (must complete in < 10 minutes)
    startTime := time.Now()

    // Launch in parallel batches of 10
    var instances []*types.Instance
    var mu sync.Mutex
    
    for batch := 0; batch < 5; batch++ {
        var wg sync.WaitGroup
        for i := 0; i < 10; i++ {
            participantID := batch*10 + i
            
            wg.Add(1)
            go func(id int) {
                defer wg.Done()
                
                instance, err := launchInstance(t, ctx, project.ID, fixtures.CreateTestInstanceOptions{
                    Template: "Python Machine Learning (Simplified)",
                    Name:     fmt.Sprintf("workshop-participant-%02d", id),
                    Size:     "S",
                })
                
                if err == nil {
                    mu.Lock()
                    instances = append(instances, instance)
                    mu.Unlock()
                }
            }(participantID)
        }
        wg.Wait()
    }

    provisionTime := time.Since(startTime)

    // Success criteria: All 50 workspaces provisioned
    assert.Len(t, instances, 50, "All workshop workspaces should provision")

    // Success criteria: Provisioning completes in < 10 minutes
    assert.Less(t, provisionTime, 10*time.Minute, "Workshop setup must complete before workshop starts")
}
```

### 2. Generate Temporary Access Credentials

```go
// Generate temporary credentials for all participants
credentials := make([]WorkshopCredential, len(instances))

for i, instance := range instances {
    // Generate time-limited SSH key (expires in 5 hours)
    cred, err := generateTemporaryCredential(t, ctx, instance.ID, 5*time.Hour)
    require.NoError(t, err)
    
    credentials[i] = WorkshopCredential{
        ParticipantID: i,
        InstanceIP:    instance.PublicIP,
        SSHKey:        cred.PrivateKey,
        JupyterURL:    fmt.Sprintf("http://%s:8888", instance.PublicIP),
        JupyterToken:  cred.JupyterToken,
        ExpiresAt:     time.Now().Add(5 * time.Hour),
    }
}

// Export credentials to CSV for distribution
exportCredentials(t, credentials, "workshop-credentials.csv")

// Verify credentials format
assert.Len(t, credentials, 50)
assert.NotEmpty(t, credentials[0].SSHKey)
assert.NotEmpty(t, credentials[0].JupyterToken)
```

### 3. During Workshop: Concurrent Access

```go
// Simulate 50 participants connecting simultaneously
var wg sync.WaitGroup
successCount := atomic.Int32{}

for i, cred := range credentials {
    wg.Add(1)
    go func(participantID int, c WorkshopCredential) {
        defer wg.Done()
        
        // Test Jupyter accessibility
        resp, err := httpGet(c.JupyterURL + "?token=" + c.JupyterToken)
        if err == nil && resp.StatusCode == 200 {
            successCount.Add(1)
        }
        
        // Test SSH connectivity
        ssh, err := connectViaSSHWithKey(c.InstanceIP, c.SSHKey)
        if err == nil {
            // Simulate participant running code
            output := ssh.Exec("python -c 'import numpy as np; print(np.array([1,2,3]))'")
            if strings.Contains(output, "[1 2 3]") {
                t.Logf("Participant %d: ML environment working", participantID)
            }
        }
    }(i, cred)
}

wg.Wait()

// Success criteria: >95% participants can access (allow for occasional failures)
accessRate := float64(successCount.Load()) / float64(len(instances)) * 100
assert.GreaterOrEqual(t, accessRate, 95.0, "At least 95% of participants should access successfully")
```

### 4. Workshop Workload Simulation

```go
// Simulate workshop exercises (concurrent notebook execution)
for i := 0; i < 10; i++ {  // Test sample of 10 participants
    ssh := connectViaSSHWithKey(instances[i].PublicIP, credentials[i].SSHKey)
    
    // Run ML training workload (simple model)
    workload := `python -c "
import numpy as np
from sklearn.linear_model import LinearRegression
X = np.random.rand(100, 1)
y = 2 * X + 1 + np.random.randn(100, 1) * 0.1
model = LinearRegression()
model.fit(X, y)
print('Model trained successfully')
"`
    
    output := ssh.Exec(workload)
    assert.Contains(t, output, "Model trained successfully")
}
```

### 5. Performance Under Load

```go
// Monitor system performance with 50 concurrent users
metrics := getProjectMetrics(t, ctx, project.ID)

// Verify no instances overloaded
for _, instance := range instances {
    instanceMetrics := getInstanceMetrics(t, ctx, instance.ID)
    
    // CPU should not be saturated (< 90%)
    assert.Less(t, instanceMetrics.CPUUtilization, 90.0, 
        "Instance %s CPU saturated: %.1f%%", instance.Name, instanceMetrics.CPUUtilization)
    
    // Memory should not be full (< 85%)
    assert.Less(t, instanceMetrics.MemoryUtilization, 85.0,
        "Instance %s memory saturated: %.1f%%", instance.Name, instanceMetrics.MemoryUtilization)
}

// Verify no API throttling
assert.Equal(t, 0, metrics.APIThrottleCount, "No AWS API throttling should occur")
```

### 6. Post-Workshop: Auto-Termination

```go
// Simulate workshop end (4 hours later)
// For testing, manually trigger auto-termination
triggerProjectAutoTermination(t, ctx, project.ID)

// Verify all instances terminate
time.Sleep(10 * time.Minute)  // Wait for graceful termination

allInstances := listInstances(t, ctx, project.ID)
for _, instance := range allInstances {
    assert.Equal(t, "terminated", instance.State, 
        "Instance %s should be terminated after workshop", instance.Name)
}

// Verify temporary credentials expired
_, err := connectViaSSHWithKey(instances[0].PublicIP, credentials[0].SSHKey)
assert.Error(t, err, "Temporary credentials should be revoked after workshop")
```

### 7. Optional: Backup Participant Work

```go
// If configured, backup all participant notebooks before termination
if config.BackupParticipantWork {
    backups := listProjectBackups(t, ctx, project.ID)
    assert.GreaterOrEqual(t, len(backups), 50, "Should backup all participant work")
    
    // Verify backup contains Jupyter notebooks
    backup := backups[0]
    backupContents := inspectBackup(t, ctx, backup.ID)
    assert.Contains(t, backupContents, ".ipynb", "Backup should contain notebooks")
}
```

### 8. Cost Verification

```go
// Verify workshop costs within budget
costSummary := getProjectCostSummary(t, ctx, project.ID)

// 50 instances * 4 hours * $0.02/hour = $4.00 (conservative estimate)
expectedCost := 50 * 4 * 0.02
assert.InDelta(t, expectedCost, costSummary.TotalSpend, 2.0, 
    "Workshop cost should be ~$4 for 50 participants * 4 hours")

// Verify cost per participant
costPerParticipant := costSummary.TotalSpend / float64(numParticipants)
assert.Less(t, costPerParticipant, 0.20, 
    "Cost per participant should be < $0.20 for 4-hour workshop")

// Verify within budget
assert.Less(t, costSummary.TotalSpend, 200.00, "Total cost should be within $200 budget")
```

## Success Criteria

- ✅ Bulk provision 50 workspaces in < 10 minutes (rapid deployment)
- ✅ Generate temporary credentials for all participants
- ✅ 95%+ participants can access Jupyter simultaneously
- ✅ ML environment works under concurrent load (no saturation)
- ✅ No AWS API throttling during bulk operations
- ✅ Auto-termination triggers 4 hours after workshop start
- ✅ Temporary credentials revoked after workshop ends
- ✅ Total workshop cost < $10 (50 participants * 4 hours)
- ✅ Optional: Participant work backed up before termination

## Test Duration

~20-25 minutes (includes bulk provisioning, concurrent access, termination)

## AWS Resources

- 50 EC2 instances (t3.small)
- 4-hour runtime (simulated or accelerated)
- Temporary IAM credentials (time-limited)
- Cost: ~$5 for test run (50 instances * 30 minutes * $0.02/hour)

## Dependencies

- `pkg/project/bulk_operations.go` - Rapid bulk provisioning
- `pkg/project/auto_termination.go` - Time-based termination
- `pkg/auth/temporary_credentials.go` - Time-limited SSH keys (TO CREATE)
- `pkg/backup/scheduler.go` - Optional pre-termination backups
- `test/fixtures/projects.go` - Project fixtures
- `test/integration/load_testing.go` - Concurrent access helpers (TO CREATE)

## Priority

🟢 **LOW-MEDIUM** - Nice to have for conference/workshop use cases. Lower priority than Solo/Lab/Class scenarios but valuable for event-based usage.

## Related Issues

- Part of Phase 2: Persona-Driven Workflows
- Follows #400 (Solo Researcher), #401 (Lab), #402 (University Class)
- Validates rapid deployment and temporary access not tested elsewhere

## Notes

- Rapid provisioning (< 10 min for 50 instances) tests AWS API rate limits
- Temporary credentials require IAM role generation with expiration
- Concurrent access (50 simultaneous users) tests system scalability
- Auto-termination critical for cost control (prevent forgetting to terminate)
- Performance monitoring ensures instances can handle workshop load
- Optional backup preserves participant work for post-workshop sharing
- Test uses 50 participants but implementation should scale to 100+
